File: Android/app/src/main/java/com/pokeapi/lpiem/pokeapiandroid/Model/Pokemon/PokemonAPI.java
package com.pokeapi.lpiem.pokeapiandroid.Model.Pokemon;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface PokemonAPI {

    @GET("/api/v2/pokemon/{id}")
    Call<Pokemon> getPokemonById(
            @Path("id") int id
    );
}

File: Android/app/src/main/java/com/pokeapi/lpiem/pokeapiandroid/Model/Retrofit/Pokemons/PokemonAPI.kt
package com.pokeapi.lpiem.pokeapiandroid.Model.Retrofit.Pokemons

import retrofit2.Call
import retrofit2.http.GET

interface PokemonAPI {

    /**
     * Fetch pokemon list for the pokedex fragment
     * @return list of pokemon name and sprites
     */
    @GET("/public/index.php/pokemon/")
    fun getPokemonListData(): Call<PokemonList>
}
